docs/understand/performance-best-practices.md

```diff
@@ -31,9 +26,61 @@ When accessing data using the branch name (e.g. `lakefs://repo/main/path`) lakeF
 For more information, see [how uncommitted data is managed in lakeFS][representing-refs-and-uncommitted-metadata].
 
 ## Operate directly on the storage
-Sometimes, storage operations can become a bottleneck. For example, when your data pipelines upload many big objects.
+Storage operations can become a bottleneck when operating on large datasets.
+
 In such cases, it can be beneficial to perform only versioning operations on lakeFS, while performing storage reads/writes directly on the object store.
-lakeFS offers multiple ways to do that:
+lakeFS offers multiple ways to do that.
+
+### Use zero-copy import
```

Comment:
I'm not sure that this is the appropriate location for this section. For example, I could import data to lakeFS and then read it directly from lakeFS. So that flow isn't "operating directly on the object store"

Reply:
+1, The zero-copy import section talks about how to regularly feed data into lakeFS rather than how to interact with data already managed by lakeFS.

Reply:
How would you prefer to organize this information?

The existing flow makes sense to me but I'm not dead set on it. 

The reason it makes sense to me is that when I looked at the LakeFS architecture diagram, the first question in my mind was "how do I bypass the lakefs service for bulk operations on data", as that's the obvious bottleneck assuming the object storage is S3 or similar. This then leads to 3 questions, which are addressed in this section:

1. If I have data already in S3, how do I make LakeFS aware of it without re-copying? Answer: use zero-copy writes / imports . (This is where DVC fell out of our evaluation, btw...)
2. If I already have a large dataset in LakeFS, how do I add  new data? 2 answers: another zero copy import is fine, or write by performing metadata operations in lakefs and writing directly to storage
3. How do I do efficient, scaleable reads? Answer: get the urls from the metadata service, talk to S3/GCS directly using the urls. 

And the fuse stuff got in here because the follow up to the third question is, what if I am not writing my own reader, but rather using fuse to mount a bucket - does this mean I'm SOL for using LakeFS at all, and if not, do all the reads go through the slow way, streaming data through the LakeFS service? And the answer to _that_ is _also_ no, LakeFS thought of that, it's all done via symlinks and you can read directly from the storage.